x-pack/plugins/task_manager/index.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TaskManager } from './task_manager';
+
+export function taskManager(kibana) {
+  return new kibana.Plugin({
+    id: 'task_manager',
+    require: ['kibana', 'elasticsearch', 'xpack_main'],
+    configPrefix: 'xpack.task_manager',
+    config(Joi) {
+      return Joi.object({
+        enabled: Joi.boolean().default(true),
+        max_attempts: Joi.number()
+          .description('The maximum number of times a task will be attempted before being abandoned as failed')
+          .default(3),
+        poll_interval: Joi.number()
+          .description('How often, in milliseconds, the task manager will look for more work.')
+          .default(3000),
+        index: Joi.string()
+          .description('The name of the index used to store task information.')
+          .default('.kibana_task_manager'),
+        max_workers: Joi.number()
+          .description('The maximum number of tasks that this Kibana instance will run simultaneously.')
+          .default(10),
```

Comment:
This number correlates to the `numWorkers` that task type definitions are created with. The number of workers is an arbitrary way to size for capacity: "1 worker" just means "the smallest size workload", and "10 workers" means "the largest size by default".

It needs to be configurable and simple to understand the effects of changing it. When you have an instance that think should only handle half as many tasks, you can just calculate `10/2` and set `max_workers` to 5.

I'm opposed to having a dynamic default for the setting, because machines that have a small number of CPUs will have small dividend for scaling down. For example, if you want the smallest task weight (1 worker) to be 1/10th of the maximum workers, you would need a multiple of 10 CPUs.